MidiPlayer/src/main/java/org/primesoft/midiplayer/midiparser/MidiParser.java

```diff
@@ -275,48 +272,44 @@ private static void setInstrument(HashMap<Integer, Instrument> instruments, int
     /**
      * Aggregate note entries based on time
      *
-     * @param notes
-     * @return
+     * @param notes The notes to aggregate
+     * @return The notes aggregated with time entries in milliseconds
      */
-    private static List<Pair<Long, HashSet<TrackEntry>>> aggregate(List<TrackEntry> notes) {
-        HashMap<Long, HashSet<TrackEntry>> tmp = new HashMap<Long, HashSet<TrackEntry>>();
+    private static List<Pair<Long, Set<TrackEntry>>> aggregate(List<TrackEntry> notes) {
+        Map<Long, Set<TrackEntry>> tmp = new HashMap<Long, Set<TrackEntry>>();
 
         for (TrackEntry entry : notes) {
-            final long milis = entry.getMilis();
+            final long millis = entry.getMillis();
 
-            final HashSet<TrackEntry> list;
-            if (tmp.containsKey(milis)) {
-                list = tmp.get(milis);
+            final Set<TrackEntry> set;
+            if (tmp.containsKey(millis)) {
```

Comment:
This part can use computeIfAbsent

Reply:
Replaced the whole if else block with a single computeIfAbsent line. Thanks for the tip, I didn't know that method.